portfolio/src/main/webapp/app.js

```diff
@@ -44,24 +44,58 @@ function getBookInfo() {
     }
   });
 }
-
-function searchBooks(book, url) {
-  $.getJSON(url, function(json) {
-     var info = json.items[0].volumeInfo;
-     var title = book.title;
-     var reviews = book.reviews;
-     var genre = book.genre;            // List of genres from GoodReads
-     var categories = info.categories;  // List of genres from Books API
-     var author = info.authors;
-     var language = info.language;
-     var description = info.description;
-     var infoLink = info.infoLink;
-     var pageCount = info.pageCount;
-     var publishedDate = info.publishedDate;
-     var publisher = info.publisher;
-     var maturityRating = info.maturityRating;
-   }).fail(function(jqxhr, status, errorMessage) {
-    $('#bookresult')
-        .html('Status Code: ' + status + '<br>Error Message: ' + errorMessage);
-  });
+ 
+const bookFields = {
+  TITLE: "title",
+  GENRE: "genre",
+  CATEGORIES: "categories",
+  AUTHOR: "author",
+  LANGUAGE: "language",
+  DESCRIPTION: "description",
+  INFO_LINK: "infoLink",
+  PAGE_COUNT: "pageCount",
+  PUBLISH_DATE: "publishedDate",
+  PUBLISHER: "publisher",
+  MATURITY_RATING: "maturityRating"
 }
+ 
+function searchBooks(book, url)
+{
+   
+   $.getJSON(url, function (json)
+   {
+        var info = json.items[0].volumeInfo;
+        var title = book.title;
+        var reviews = book.reviews;
+        var genre = book.genre; // List of genres from GoodReads
+        var categories = info.categories; // List of genres from Books API
+        var author = info.authors;
+        var language = info.language;
+        var description = info.description; 
+        var infoLink = info.infoLink;
+        var pageCount = info.pageCount;
+        var publishedDate = info.publishedDate;
+        var publisher = info.publisher;
+        var maturityRating = info.maturityRating;
+ 
+        var fieldContentArray = [title, genre, categories, author, language, description, infoLink, pageCount, publishedDate, publisher, maturityRating];
+        var fieldsString;
+ 
+        //passing values through as URL parameters
+        for (var field in bookFields){
+           if(index == 0){
+             fieldsString = "?"+field+"="+fieldContentArray[index];
```

Comment:
Does this code actually work? I imagine it would throw errors when trying to evaluate 'index', since it isn't defined anywhere.

Reply:
I changed the looping method. The syntax was off, now it is working properly.